optashift-employee-rostering-gwtui/src/main/java/org/optaplanner/openshift/employeerostering/gwtui/client/rostergrid/view/ViewportView.java

```diff
@@ -44,16 +53,69 @@
     @Inject
     private TimingUtils timingUtils;
 
+    private MutationObserver domObserver;
+
+    private MutationObserverInit domObserverConfig;
+
+    // In an ideal world we wouldn't need these, but alas this isn't an ideal world
+    private static native void addClassToElement(Object element, String className) /*-{
+        if (element.classList) {
+            element.classList.add(className);
+            var children = element.children;
+            for (var i = 0; i < children.length; i++) {
+                @org.optaplanner.openshift.employeerostering.gwtui.client.rostergrid.view.ViewportView::addClassToElement(Ljava/lang/Object;Ljava/lang/String;)(children[i],className);
+            }
+        }
+    }-*/;
```

Comment:
We might be able to remove this method and live in a better world 😛. Check out this gist I made with a proposal to remove it https://gist.github.com/tiagobento/e1c7edbbd0e8fe378f2b01151a37b9f6. Wdyt?

Reply:
99% sure this won't work since "MutationObserver" addedNodeList  is not of elemental Node type, but of GWT node type (so we get a typecast exception when you are creating the list)

Reply:
(actually that unchecked cast might fix the issue, let me check)

Reply:
It's not a problem to cast a GWT Node to an Elemental2 Node, they represent the  same object in JS. They just have a different Java API.

Reply:
It works; I think I tried that with `Js.cast` instead of `Js.uncheckedCast`, hence my issue